Add documents about controllers inherited from kubernetes. (#1558)
karmada-io/karmada: docs/userguide/configure-controllers.md

**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*

- [Configure Controllers](#configure-controllers)
  - [Karmada Controllers](#karmada-controllers)
    - [Configure Karmada Controllers](#configure-karmada-controllers)

>   - [Kubernetes controllers](#kubernetes-controllers)

Reviewer: ```suggestion
  - [Kubernetes Controllers](#kubernetes-controllers)
```

>     - [Recommended controllers](#recommended-controllers)

Reviewer: ```suggestion
    - [Recommended Controllers](#recommended-controllers)
```

Use `-foo` to disable the controller named `foo`.

> > Note: The default controller list might be changed in the following release. The controllers enabled by last release

Reviewer: ```suggestion
> Note: The default controller list might be changed in the future releases. The controllers enabled in the last release
```

> might be disabled or deprecated and new controllers might be introduced too. Users who are using this flag should
> check the release notes before system upgrade.

## Kubernetes controllers

In addition to the controllers that are maintained by the Karmada community, Karmada also requires some controllers from
Kubernetes. These controllers run as part of `kube-controller-manager` and are maintained by the Kubernetes community.

Users are recommended to deploy the `kube-controller-manager` along with Karmada components. And the installation
methods list in [installation guide][2] would help you deploy it as well as Karmada components.

### Recommended controllers

Not all controllers in `kube-controller-manager` are necessary for Karmada, if you are deploying

> Karmada by other tools, you might have to configure the controllers by `--controllers` flag just like what we did in

Reviewer: ```suggestion
Karmada using other tools, you might have to configure the controllers by `--controllers` flag just like what we did in
```

[example of kube-controller-manager deployment][3].

> Following controllers are tested and recommended by Karmada.

Reviewer: ```suggestion
The following controllers are tested and recommended by Karmada.
```

#### namespace
TODO.

#### garbagecollector
TODO.

#### serviceaccount-token

> The `serviceaccount-token` controller runs as part of `kube-controller-manager`, one of its responsibility is

Reviewer: ```suggestion
The `serviceaccount-token` controller runs as part of `kube-controller-manager`.
```

> it watches `ServiceAccount` creation and creates a corresponding ServiceAccount token Secret to allow API access.

Reviewer: ```suggestion
It watches `ServiceAccount` creation and creates a corresponding ServiceAccount token Secret to allow API access.
```
